docs/tutorials/csharp/bert-nlp-csharp-console-app.md

```diff
@@ -0,0 +1,376 @@
+    logging.info("Post-processing")  
+
+    # Post process the output of the model into an answer (or an error if the question could not be answered)
+    results = postprocess(tokens, outputs[0], outputs[1])
+    logging.info(results)
+    return results
+
+```
+- Call the `init()` function to download the model and create the `InferenceSession`.
+
+```python
+init()
+```
+- Create the `input` and run the model. 
+```python
+input = "{\"question\": \"What is Dolly Parton's middle name?\", \"context\": \"Dolly Rebecca Parton is an American singer-songwriter\"}"
+
+print(run(input))
+```
+- Here is what the output should look like for the above question. Use the `input_ids` to validate the tokenization in C#.
+
+```text
+Output:
+{'input_ids': [101, 2054, 2003, 19958, 2112, 2239, 1005, 1055, 2690, 2171, 1029, 102, 19958, 9423, 2112, 2239, 2003, 2019, 2137, 3220, 1011, 6009, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
+{'answer': 'Rebecca'}
+```
+## Inference with C#
+
+Now that we have tested the model in Python its time to build it out in C#. The first thing we need to do is to create our project. For this example we will be using a Console App however you could use this code in any C# application.
+
+- Open Visual Studio and [Create a Console App](https://docs.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console?view=vs-2022)
+
+### Install the Nuget Packages
+- Install the Nuget packages `BERTTokenizers`, `Microsoft.ML.OnnxRuntime`, `Microsoft.ML.OnnxRuntime.Managed`, `Microsoft.ML`
+```PowerShell
+dotnet add package Microsoft.ML.OnnxRuntime --version 1.12.0
+dotnet add package Microsoft.ML.OnnxRuntime.Managed --version 1.12.0
+dotnet add package dotnet add package Microsoft.ML
+dotnet add package dotnet add package BERTTokenizers --version 1.1.0
+```
+
+### Create the App
+
+- Import the packages
+
+```csharp
+using BERTTokenizers;
+using Microsoft.ML.Data;
+using Microsoft.ML.OnnxRuntime;
+using Microsoft.ML.OnnxRuntime.Tensors;
+using System;
+```
+- Add the `namespace`, `class` and `Main` function.
+
+```csharp
+
+namespace MyApp // Note: actual namespace depends on the project name.
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+
+        }
+    }
+}
+```
+### Create the BertInput class for encoding
+- Add the `BertInput` class
+
+```csharp
+    public class BertInput
+    {
+        public long[] InputIds { get; set; }
+        public long[] AttentionMask { get; set; }
+        public long[] TypeIds { get; set; }
+    }
+```
+### Tokenize the sentence with the `BertUncasedLargeTokenizer`
+- Create a sentence (question and context) and tokenize the sentence with the `BertUncasedLargeTokenizer`. The base model for this fine tuned model was the BERT Uncased Large so the tokenizer is the same.
```

Comment:
This is not super clear. Would all fine tuned BERT models use this tokenizer? If so, just say that. If not, maybe point when you would use this and when you wouldn't

Reply:
There are different base models and tokenizers that a BERT model may have originated from. Based on what the model was fine tuned from indicates with tokenizer to use. I explain that a bit up earlier in the tutorial on line 25.

Reply:
So would something like: 

`Create a sentence (question and context) and tokenize the sentence with the `BertUncasedLargeTokenizer`. This tokenizer can used for all BERT based HuggingFace models.`